test/firebase_auth_mocks_test.dart

```diff
@@ -763,16 +792,16 @@ void main() {
     expect(decodedToken['bodyHeight'], 169);
   });
 
-  test('The customClain should exist after sign-out and sign-in', () async {
-    final auth = MockFirebaseAuth(
-      mockUser: MockUser(customClaim: {'role': 'admin', 'bodyHeight': 169}),
-      signedIn: true,
-    );
+  test('The customClaim should exist after sign-out and sign-in', () async {
+    final user = MockUser(customClaim: {'role': 'admin', 'bodyHeight': 169});
+    final auth = MockFirebaseAuth(mockUser: user, signedIn: true);
     final decodedToken =
         JwtDecoder.decode((await auth.currentUser!.getIdToken())!);
     expect(decodedToken['role'], 'admin');
     expect(decodedToken['bodyHeight'], 169);
     await auth.signOut();
+
+    auth.mockUser = user;
```

Comment:
I think I have misnamed `mockUser`. It seems like `mockUser` is the user for the next signin, but, if signed in, also lets you set the current user (https://github.com/atn832/firebase_auth_mocks/pull/80).

In the future, renaming `mockUser` to `userOnNextSignIn`, and exposing the currently signed in user to allow #80 's use case might be more appropriate.

I need to think about it more before changing the API and breaking people's projects.

Reply:
If I think of `mockUser` as the `userOnNextSignIn`, it feels strange that signing out should set it to null. It will also break people's projects if they sign out and sign in again, like in this test.

Since you've allowed setting mockUser to null, would that be enough? Then you can implement your use case as :

```dart
auth.signIn(...);
auth.mockUser = null;
auth.signInAnonymously();
```

It's not perfect but it won't break any existing project until we figure out a better solution.

Reply:
No problem - I definitely think my understanding of how `mockUser` is intended to be used was lacking. I am happy to revert some changes here as you suggest.

I need to specifically look at how we handle the case in #110 as per your suggestion though, as I'm not sure we have that fine grained control. I will check back shortly.

Reply:
@atn832 Having revisited our problem with your explanation of `mockUser`, I think your suggestion makes sense and works for us. I've reverted the change to clear the mock user now, and left in everything else. Let me know what you think.